config/config.go

```diff
@@ -143,6 +143,46 @@ func InitChains(ctx *Context, homePath string, debug bool) error {
 	return nil
 }
 
+func (c *Config) UnmarshalConfig(homePath, configPath string) error {
+	cfgPath := fmt.Sprintf("%s/%s", homePath, configPath)
+	if _, err := os.Stat(cfgPath); err == nil {
+		file, err := os.ReadFile(cfgPath)
+		if err != nil {
+			return err
+		}
+		// unmarshall them into the struct
+		if err = UnmarshalJSON(file, c); err != nil {
+			return err
+		}
+	} else {
+		defConfig := defaultConfig(cfgPath)
+		*c = defConfig
+	}
+	c.ConfigPath = cfgPath
+	return nil
+}
+
+func (ctx *Context) InitConfig(homePath string, debug bool) error {
```

Comment:
Old function signature `(c *Config) InitConfig(ctx *Context, ...)` is a bit confusing that c and ctx.Config should be same pointer or not.